### ISSUE TITLE
./configure --disable-ncurses fails to link `ftptop`: undefined reference to symbol 'stdscr'

### ISSUE THREAD

**Issue by trofi** [proftpd/proftpd#1207]
It's an upstream version of downstream https://bugs.gentoo.org/777531 report. Gentoo uses split `-lncursesw/-ltinfow` library (and also `-lncurses/-ltinfo`). Build failure seems to be easily reproducible on current git as:

```
$ ./configure --disable-ncurses && make
$ make
$ /bin/sh ./libtool --mode=link --tag=CC x86_64-pc-linux-gnu-gcc -L./lib -L./lib  -rdynamic   -o proftpd src/main.o src/timers.o src/sets.o src/pool.o src/privs.o src/str.o src/table.o src/regexp.o src/configdb.o src/dirtree.o src/expr.o src/signals.o src/support.o src/netaddr.o src/inet.o src/child.o src/parser.o src/log.o src/lastlog.o src/xferlog.o src/bindings.o src/netacl.o src/class.o src/scoreboard.o src/help.o src/feat.o src/netio.o src/cmd.o src/response.o src/ascii.o src/data.o src/modules.o src/stash.o src/display.o src/auth.o src/fsio.o src/mkhome.o src/ctrls.o src/event.o src/var.o src/throttle.o src/session.o src/trace.o src/encode.o src/proctitle.o src/filter.o src/pidfile.o src/env.o src/random.o src/version.o src/rlimit.o src/wtmp.o src/json.o src/jot.o src/memcache.o src/redis.o src/error.o modules/mod_core.o modules/mod_xfer.o modules/mod_rlimit.o modules/mod_auth_unix.o modules/mod_auth_file.o modules/mod_auth.o modules/mod_ls.o modules/mod_log.o modules/mod_site.o modules/mod_delay.o modules/mod_facts.o modules/mod_auth_pam.o  modules/mod_cap.o modules/module_glue.o  -lsupp -lattr -lcrypt -ldl   -lcap  -lpam -ldl  -lidn `uniq ./module-libs.txt | tr '\n' ' '`
libtool: link: x86_64-pc-linux-gnu-gcc -rdynamic -o proftpd src/main.o src/timers.o src/sets.o src/pool.o src/privs.o src/str.o src/table.o src/regexp.o src/configdb.o src/dirtree.o src/expr.o src/signals.o src/support.o src/netaddr.o src/inet.o src/child.o src/parser.o src/log.o src/lastlog.o src/xferlog.o src/bindings.o src/netacl.o src/class.o src/scoreboard.o src/help.o src/feat.o src/netio.o src/cmd.o src/response.o src/ascii.o src/data.o src/modules.o src/stash.o src/display.o src/auth.o src/fsio.o src/mkhome.o src/ctrls.o src/event.o src/var.o src/throttle.o src/session.o src/trace.o src/encode.o src/proctitle.o src/filter.o src/pidfile.o src/env.o src/random.o src/version.o src/rlimit.o src/wtmp.o src/json.o src/jot.o src/memcache.o src/redis.o src/error.o modules/mod_core.o modules/mod_xfer.o modules/mod_rlimit.o modules/mod_auth_unix.o modules/mod_auth_file.o modules/mod_auth.o modules/mod_ls.o modules/mod_log.o modules/mod_site.o modules/mod_delay.o modules/mod_facts.o modules/mod_auth_pam.o modules/mod_cap.o modules/module_glue.o  -L/home/slyfox/dev/git/proftpd/lib -lsupp /usr/lib64/libattr.so -lcrypt -lcap -lpam -ldl -lidn
cd utils/ && make utils
make[1]: Entering directory '/home/slyfox/dev/git/proftpd/utils'
make[1]: Nothing to be done for 'utils'.
make[1]: Leaving directory '/home/slyfox/dev/git/proftpd/utils'
x86_64-pc-linux-gnu-gcc -L./lib -L./lib  -rdynamic  -o ftptop utils/ftptop.o utils/scoreboard.o utils/misc.o -lncurses -lsupp -lattr -lcrypt -ldl
/usr/lib/gcc/x86_64-pc-linux-gnu/11.0.1/../../../../x86_64-pc-linux-gnu/bin/ld: utils/ftptop.o: undefined reference to symbol 'stdscr'
/usr/lib/gcc/x86_64-pc-linux-gnu/11.0.1/../../../../x86_64-pc-linux-gnu/bin/ld: /lib64/libtinfo.so.6: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make: *** [Makefile:95: ftptop] Error 1
```

```
$ cat Make.rules
...
CURSES_LIBS=-lncurses
UTILS_LIBS= -lsupp -lattr -lcrypt -ldl
...
```

AFAIU `UTILS_LIBS` should have contained `-ltinfo`.

**Comment by thesamesam** [comment]
Is the intention to build ftptop (which requires tinfo) based on --enable-ncurses or unconditionally (and therefore always check for tinfo(w), which is a bit funky)?

**Comment by Castaglia** [comment]
@thesamesam the intention is to build `ftptop` unconditionally -- so that it can be executed, and the runtime will print an error to the user describing the reason why it can't work (if no curses/tinfo support is available), for a better user experience.

**Comment by Castaglia** [comment]
This may also be related to Issue #1174, which pertains to `ftptop` on Gentoo.

**Comment by Castaglia** [comment]
@trofi Are you able to reproduce this issue using the latest code from `master`, or from the 1.3.7 branch -- which contain the fix for Issue #1174? I'm currently attempting to do so...

**Comment by trofi** [comment]
The above session was against yesterday's git master. Updated today just in case and got the same failure on 89a1e6d741686eb6b5f1f11d26976d49fcdae515.

Attaching [config.log](https://github.com/proftpd/proftpd/files/6187685/config.log) to ease comparing detected libraries.

**Comment by Castaglia** [comment]
@trofi got it, thanks!  I tend to develop on Ubuntu/Mac, sometimes Centos; I didn't see the above error there (probably because those platforms may not have the `libncurses/libtinfo` split?)  Happen to know of a Docker image I might use, to reproduce (and then fix/verify) this issue locally?

**Comment by Castaglia** [comment]
One thing that stands out to me, in the provided config.log, is that `configure` still looks for the curses/ncurses libraries -- and add them to the Make.rules if found -- regardless of the use of `--disable-ncurses`.  Based on that, I've opened https://github.com/proftpd/proftpd/pull/1209; I'm curious to see how that would affect your build.

**Comment by trofi** [comment]
I did not use it personally but we apparently have official images: https://hub.docker.com/r/gentoo/stage3.

**Comment by Castaglia** [comment]
Using this Dockerfile:
```
FROM gentoo/portage:latest as portage

FROM gentoo/stage3-amd64:latest

COPY --from=portage /var/db/repos/gentoo /var/db/repos/gentoo

CMD ["bash"]
```
I was able to reproduce the build error (although for a different symbol):
```
gcc -L./lib -L./lib  -rdynamic  -o ftptop utils/ftptop.o utils/scoreboard.o utils/misc.o -lncurses -lsupp -lattr -lcrypt -ldl 
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: utils/ftptop.o: undefined reference to symbol 'nodelay'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /lib64/libtinfo.so.6: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

**Comment by Castaglia** [comment]
Well, this is certainly unexpected, but does explain the build error even when `--disable-ncurses` is used:
```
# ls -al /usr/lib64/libcurses.so 
lrwxrwxrwx 1 root root 13 Apr 10 18:36 /usr/lib64/libcurses.so -> libncurses.so
```

**Comment by Castaglia** [comment]
This should now be fixed in master; the fix was backported to the 1.3.7 branch.  Thanks!

**Comment by trofi** [comment]
I can confirm master branch builds without problems for me now. Thank you!
